internal/restorer/restorer.go

```diff
@@ -598,6 +599,12 @@ func (res *Restorer) VerifyFiles(ctx context.Context, dst string) (int, error) {
 		work     = make(chan mustCheck, 2*nVerifyWorkers)
 	)
 
+	if p != nil {
+		numFilesVerified := uint64(res.sn.Summary.TotalFilesProcessed)
```

Comment:
I just had another look and we'll need a different way to determine the number of files to verify. Just restore the same snapshot twice, then you'll end up with something like the following:
```
verifying files in output
[0:00] 0.00%  0 / 1 files verified
finished verifying 0 files in output (took 0s)
```

This is rather confusing as the progress will never reach 100% in most cases. Or even worse, just restore an old snapshot without that metadata and restic would just crash.

What could work is the following. In `RestoreTo` there's a `res.trackFile(location, updateMetadataOnly)` call. At that place the restorer could also count all files that are not `updateMetadataOnly` updates. Then `RestoreTo` could return that number similar to `VerifyFiles`.

That will also remove the hacks that were necessary to make the tests work.

Reply:
Thanks for the catching this, and the guidance!

I've updated the code as per your suggestions.

Please review again when you have the time :)